src/core/CoolBoardSensors.cpp

```diff
@@ -206,15 +206,21 @@ float CoolBoardSensors::readSoilMoisture() {
   digitalWrite(MOISTURE_SENSOR_PIN, LOW);
   digitalWrite(ANALOG_MULTIPLEXER_PIN, HIGH);
   delay(2000);
-
-  // read moisture sensor value
-  int val = analogRead(A0);
-
-  if (val >= 891) {
-    val = 890;
+  int val = 0;
+  // oversample value 64 times for stable readings
+  for (int i = 0; i < 64; i++) {
+    delay(10);
+    val = val + analogRead(A0);
+    delay(10);
   }
-
-  float result = (float)map(val, 0, 890, 0, 100);
+  // divide by 64 to get the average
+  val=val/64;
+  float x_deux = 0.00027;
```

Comment:
It's been a while since I've last seen code written in french :)

The use of english is mandatory. It is less verbose, more open-source friendly and a common coding practice among the larger dev community.